wefe/debias/half_sibling_regression.py

```diff
@@ -0,0 +1,351 @@
+    ) -> WordEmbeddingModel:
+        """Substracts the gender information from vectors.
+
+        Parameters
+        ----------
+        model : WordEmbeddingModel
+            The word embedding model to mitigate.
+        target : Optional[List[str]], optional
+            If a set of words is specified in target, the debias method
+            will be performed only on the word embeddings of this set.
+            If `None` is provided, the debias will be performed on all
+            non gender specific words (except those specified in ignore).
+            Target words must not be included in the gender specific set.
+            by default `None`.
+        ignore : Optional[List[str]], optional
+            If target is `None` and a set of words is specified in ignore,
+            the debias method will perform the debias in all non gender
+            specific words except those specified in this
+            set, by default `[]`.
+        copy : bool, optional
+                If `True`, the debias will be performed on a copy of the
+                model.
+                If `False`, the debias will be applied on the same model
+                delivered, causing its vectors to mutate.
+                **WARNING:** Setting copy with `True` requires RAM at least
+                2x of the size of the model, otherwise the execution of the
+                debias may raise to `MemoryError`, by default True.
+
+        Returns
+        -------
+        WordEmbeddingModel
+            The debiased embedding model.
+        """
+        # check if the following attributes exist in the object.
+        check_is_fitted(
+            self, ["bias_definitional_words", "non_bias", "alpha", "non_bias_dict"],
+        )
+
+        if self.verbose:
+            print(f"Executing Half Sibling Debias on {model.name}")
+
+        # -------------------------------------------------------------------
+        # Copy
+        if copy:
+            print(
+                "Copy argument is True. Transform will attempt to create a copy "
+                "of the original model. This may fail due to lack of memory."
+            )
+            model = deepcopy(model)
+            print("Model copy created successfully.")
+
+        else:
+            print(
+                "copy argument is False. The execution of this method will mutate "
+                "the original model."
+            )
+
+        # -------------------------------------------------------------------
+        # Substract bias information from vectors:
+
+        if self.verbose:
+            print("Subtracting bias information.")
+        # if target or ignore are specified the debias is applied only in the
+        # columns corresponding to those words embeddings
+        if target or ignore:
+            if target:
+                target = list(set(target) - set(ignore))
+            else:
+                target = list(set(list(self.non_bias_dict.keys())) - set(ignore))
```

Comment:
This code will crash when Ignore is None, since you are going to try to create a set from None.

Also, according to the definition of target (if target is defined, then debias is applied only on that set), you should not consider ignore since it should always be None in that case.

Reply:
In fact, I think there is a test that is failing because of this.

Reply:
fixed!